exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/text/compliant/TextReader.java

```diff
@@ -231,33 +231,34 @@ private void parseQuotedValue(byte prev) throws IOException {
     final TextInput input = this.input;
     final byte quote = this.quote;
 
-    ch = input.nextChar();
+    try {
+      input.setMonitorForNewLine(false);
+      ch = input.nextChar();
 
-    while (!(prev == quote && (ch == delimiter || ch == newLine || isWhite(ch)))) {
-      if (ch != quote) {
-        if (prev == quote) { // unescaped quote detected
-          if (parseUnescapedQuotes) {
-            output.append(quote);
-            output.append(ch);
-            parseQuotedValue(ch);
-            break;
-          } else {
-            throw new TextParsingException(
-                context,
-                "Unescaped quote character '"
-                    + quote
-                    + "' inside quoted value of CSV field. To allow unescaped quotes, set 'parseUnescapedQuotes' to 'true' in the CSV parser settings. Cannot parse CSV input.");
+      while (!(prev == quote && (ch == delimiter || ch == newLine || isWhite(ch)))) {
+        if (ch != quote) {
+          if (prev == quote) { // unescaped quote detected
+            if (parseUnescapedQuotes) {
+              output.append(quote);
+              output.append(ch);
+              parseQuotedValue(ch);
+              break;
+            } else {
+              throw new TextParsingException(context, "Unescaped quote character '" + quote + "' inside quoted value of CSV field. To allow unescaped quotes, set 'parseUnescapedQuotes' to 'true' in the CSV parser settings. Cannot parse CSV input.");
+            }
           }
+          output.append(ch);
+          prev = ch;
+        } else if (prev == quoteEscape) {
+          output.append(quote);
+          prev = NULL_BYTE;
+        } else {
+          prev = ch;
         }
-        output.append(ch);
-        prev = ch;
-      } else if (prev == quoteEscape) {
-        output.append(quote);
-        prev = NULL_BYTE;
-      } else {
-        prev = ch;
+        ch = input.nextChar();
       }
-      ch = input.nextChar();
+    } finally {
```

Comment:
I see why it is done in finally. However, as noted above, I'm not sure that pushing this kind of flag into the getChar function is the optimal approach...